src/NHibernate/Dialect/Oracle10gDialect.cs

```diff
@@ -26,12 +26,12 @@ public override JoinFragment CreateOuterJoinFragment()
 
 		public override void Configure(IDictionary<string, string> settings)
 		{
-			base.Configure(settings);
-
 			_useBinaryFloatingPointTypes = PropertiesHelper.GetBoolean(
 				Environment.OracleUseBinaryFloatingPointTypes,
 				settings,
 				false);
+
+			base.Configure(settings);
```

Comment:
Method `RegisterFloatingPointTypeMappings` is called inside `base.Configure`, so the field must be set before calling it.